lib/WebService/Async/Desk.pm

```diff
@@ -308,7 +308,7 @@ sub paging {
         }, sub {
             my ($err, @details) = @_;
             $log->errorf('Failed to request %s: %s', $uri, $err);
-            $src->completed->fail($err, @details) unless $src->completed->is_ready;
+            $src->fail($err, @details) unless $src->is_ready;
```

Comment:
I guess it should be `$f->fail(...)`

Reply:
I think what we want to mark as fail is the `$src` not `$f` which is `$src->completed`